### PR TITLE
Xenos once again block marines diagonally

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

https://github.com/tgstation/TerraGov-Marine-Corps/assets/22431091/e7257b5d-2a1b-4d21-9a9a-b1151371ec46
## Why It's Good For The Game

Marines can kinda do some still escapes currenly and it can be pretty hard for xenos to do something to stop it in certain cases.

NVIDIA_Share_Z7Em4zZomG
For example the marine can just walk like this which isn't ideal.

This also restores behavior to the old one from before my diagonal PRs.
## Changelog
:cl:
balance: Marines cannot skip past xenos diagonally if there's an obstacle between them, check the PR for a more clear explanation
/:cl:

### CODE REVIEW

**Comment by yyzsong** [comment]
We're fucking back

**Comment by Helg2** [comment]
Skill issue

**Comment by fe-ed** [comment]
THE WAR IS NEAR

**Comment by Barnet2** [comment]
Thank god, mazes are real once again

**Comment by chizzy376** [comment]
Honey the xeno winrates are down again

**Comment by Wisemonster** [comment]
This will definitely make mazes more deadly, and less of something that's just "in the way" or xeno cades.

**Comment by novaepee** [comment]
darth-vader-no-3136937808

**Comment by SimplySmiley123** [comment]
> Make it so marines block xeno from moving diagonally to when the xenos are on harm intent so it's even

This.
Punish xenomorphs for overextending as well.

**Comment by Tommy-211** [comment]
https://github.com/tgstation/TerraGov-Marine-Corps/assets/149253603/da44024d-a47a-48a9-8c61-97293ea4501c

**Comment by MiniMeatwad** [comment]
Makes sense to be one sided, it's the melee-focused team. Use a jetpack if you wanna avoid it altogether though.

**Comment by SimplySmiley123** [comment]
> Makes sense to be one sided, it's the melee-focused team. Use a jetpack if you wanna avoid it altogether though.

Nah I just want xenomorphs to get punished for overextending themselves as well, makes sense for both sides imo to have difficulties with this.
You already have abilities that knock over marines, using those for escapes would be neat imo to spice things up and give something to think about beyond "run in, use abilities, run out"

**Comment by ivanmixo** [comment]
Marine main shamelessly tries to prod me into making a xeno buff into the biggest xeno nerf in years

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -327,8 +327,10 @@
 				mob_swap_mode = PHASING
 			else if((move_resist >= MOVE_FORCE_VERY_STRONG || move_resist > L.move_force) && a_intent == INTENT_HELP) //Larger mobs can shove aside smaller ones. Xenos can always shove xenos
 				mob_swap_mode = SWAPPING
-			///if we're moving diagonally, but the mob isn't on the diagonal destination turf we have no reason to shuffle/push them
-			if(moving_diagonally && (get_dir(src, L) in GLOB.cardinals) && get_step(src, dir).Enter(src, loc))
+			/* If we're moving diagonally, but the mob isn't on the diagonal destination turf and the destination turf is enterable we have no reason to shuffle/push them
+			 * However we also do not want mobs of smaller move forces being able to pass us diagonally if our move resist is larger, unless they're the same faction as us
+			*/
+			if(moving_diagonally && (get_dir(src, L) in GLOB.cardinals) && (L.faction == faction || L.move_force <= move_resist) && get_step(src, dir).Enter(src, loc))
```
Comment (QualityVan) [inline code/modules/mob/living/living.dm:333]: Comparing the moving mob's move_resist to the standing mob's move_force seems backwards. Is it just weird naming?

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**